packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -122,178 +114,177 @@ export default {
 
     const fieldsToSelect = uniq(['id', mainField, PUBLISHED_AT_ATTRIBUTE]);
 
+    return {
+      attribute,
+      fieldsToSelect,
+      targetedModel,
+      mainField,
+      modelSchema,
+      targetField,
+    };
+  },
+
+  async findAvailable(ctx: any) {
+    await validateFindAvailable(ctx.request.query);
+
+    const { id } = ctx.request.query;
+
+    const validation = await this.extractAndValidateRequestInfo(ctx, id);
+    if (!validation) {
+      // If validation of the request has failed the error has already been sent
+      // to the ctx
+      return;
+    }
```

Comment:
I would throw errors instead of sending the context to add them to the body

```ts
import { errors } from "@strapi/types"

...

throw new errors.NotFound()
```